src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerActionCounts.cs

```diff
@@ -82,6 +82,11 @@ internal class AnalyzerActionCounts
                 OperationBlockActionsCount > 0 ||
                 OperationBlockStartActionsCount > 0 ||
                 SymbolStartActionsCount > 0;
+
+            HasAnyActionsRequiringSemanticAnalysis = HasAnyExecutableCodeActions ||
```

Comment:
@Youssef1313 first property HasAnyExecutableCodeActions includes the SyntaxNodeActionCounts case

Reply:
@mavasani Thanks for clarification. What about actions that have access to `Compilation` and can ask for symbols (common) or get a semantic model (not common)? These include additional file and compilation start actions?

Reply:
i would def comment this code :)